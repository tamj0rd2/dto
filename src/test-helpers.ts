import { resolve } from 'path'
import { readdirSync } from 'fs'

export const runTests = (testFolder: string, testPath: string) => {
  const fixturePrefix = testPath.replace(`${testFolder}/`, '').replace('.spec.ts', '')
  const files = readdirSync(resolve(process.cwd(), 'src', '__fixtures__', fixturePrefix))

  files.forEach((fileName) => {
    test(fileName, () => {
      expect(`${fixturePrefix}/${fileName}`).toMatchDiagnosticsSnapshot()
    })
  })
}
