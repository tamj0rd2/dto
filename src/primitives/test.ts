import { createTestCase } from '../test-helpers'

describe('Primitives', () => {
  const fixtures = ['null', 'boolean', 'number', 'string']

  fixtures.forEach((fixture) => {
    it(`returns correct diagnostics for ${fixture} cases`, () => {
      expect(createTestCase(__dirname, `${fixture}.ts`)).toMatchDiagnosticsSnapshot()
    })
  })
})
