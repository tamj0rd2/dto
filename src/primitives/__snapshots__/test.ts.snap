// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primitives returns correct diagnostics for boolean cases 1`] = `
Array [
  Object {
    "character": 14,
    "fileName": "boolean.ts",
    "line": 5,
    "message": "Type 'null' is not assignable to type 'boolean'.",
  },
  Object {
    "character": 14,
    "fileName": "boolean.ts",
    "line": 7,
    "message": "Type 'null' is not assignable to type 'boolean'.",
  },
  Object {
    "character": 14,
    "fileName": "boolean.ts",
    "line": 9,
    "message": "Type 'string' is not assignable to type 'boolean'.",
  },
  Object {
    "character": 14,
    "fileName": "boolean.ts",
    "line": 11,
    "message": "Type '{ bad: string; }' is not assignable to type 'boolean'.",
  },
]
`;

exports[`Primitives returns correct diagnostics for null cases 1`] = `
Array [
  Object {
    "character": 14,
    "fileName": "null.ts",
    "line": 5,
    "message": "Type 'undefined' is not assignable to type 'null'.",
  },
  Object {
    "character": 14,
    "fileName": "null.ts",
    "line": 7,
    "message": "Type '123' is not assignable to type 'null'.",
  },
  Object {
    "character": 14,
    "fileName": "null.ts",
    "line": 9,
    "message": "Type '{ hey: string; }' is not assignable to type 'null'.",
  },
]
`;

exports[`Primitives returns correct diagnostics for number cases 1`] = `
Array [
  Object {
    "character": 14,
    "fileName": "number.ts",
    "line": 5,
    "message": "Type 'null' is not assignable to type 'number'.",
  },
  Object {
    "character": 14,
    "fileName": "number.ts",
    "line": 7,
    "message": "Type 'null' is not assignable to type 'number'.",
  },
  Object {
    "character": 14,
    "fileName": "number.ts",
    "line": 9,
    "message": "Type 'string' is not assignable to type 'number'.",
  },
  Object {
    "character": 14,
    "fileName": "number.ts",
    "line": 11,
    "message": "Type '{ bad: string; }' is not assignable to type 'number'.",
  },
]
`;

exports[`Primitives returns correct diagnostics for string cases 1`] = `
Array [
  Object {
    "character": 14,
    "fileName": "string.ts",
    "line": 5,
    "message": "Type 'number' is not assignable to type 'string'.",
  },
  Object {
    "character": 14,
    "fileName": "string.ts",
    "line": 7,
    "message": "Type 'undefined' is not assignable to type 'string'.",
  },
  Object {
    "character": 14,
    "fileName": "string.ts",
    "line": 9,
    "message": "Type 'boolean' is not assignable to type 'string'.",
  },
  Object {
    "character": 14,
    "fileName": "string.ts",
    "line": 11,
    "message": "Type 'null' is not assignable to type 'string'.",
  },
  Object {
    "character": 14,
    "fileName": "string.ts",
    "line": 13,
    "message": "Type '{ hello: string; }' is not assignable to type 'string'.",
  },
]
`;
